src/posix/threading.c

```diff
@@ -131,20 +131,35 @@ void sys_rwlock_unlock(sys_rwlock_t *lock)
     assert(result == 0);
 }
 
+#define SYS_THREAD_NAME_MAX 15  // thread name may be at most 15 bytes
+
+static const char thread_names[SYS_THREAD_ROLE_COUNT][SYS_THREAD_NAME_MAX + 1] = {
+    // Indexed by sys_thread_role_t. pthread names must be < 15 chars.
+    "main",          // SYS_THREAD_MAIN - root thread (not pthread!)
+    "core_thread",   // SYS_THREAD_CORE
+    "wrkr_",         // SYS_THREAD_PROACTOR (multiple)
+    "vflow_thread",  // SYS_THREAD_VFLOW
+    "lws_thread"     // SYS_THREAD_LWS_HTTP
+};
+
+static sys_atomic_t proactor_thread_count = 0;
 
 struct sys_thread_t {
     pthread_t thread;
     void *(*f)(void *);
     void *arg;
-    char  name[16];
+    void                      *proactor_context;
+    sys_thread_proactor_mode_t proactor_mode;
+    sys_thread_role_t          role;
+    char                       name[SYS_THREAD_NAME_MAX + 1];
 };
 
 // initialize the per-thread _self to a non-zero value.  This dummy value will
 // be returned when sys_thread_self() is called from the process's main thread
 // of execution (which is not a pthread).  Using a non-zero value provides a
 // way to distinguish a thread id from a zero (unset) value.
 //
-static sys_thread_t           _main_thread_id = {.name = "main"};
+static sys_thread_t           _main_thread_id = {.role = SYS_THREAD_MAIN, .name = "main"};
```

Comment:
This question is not related to this PR. Why is this called _main_thread_id and not just _main_thread ?

Reply:
Rename done.